subsys/bluetooth/mesh/shell/shell_utils.c

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <bluetooth/mesh/models.h>
+#include <shell/shell.h>
+#include <ctype.h>
+#include <string.h>
+
+#include "mesh/net.h"
+#include "mesh/access.h"
+#include "shell_utils.h"
+
+struct shell_model_instance {
+	uint16_t addr;
+	uint8_t elem_idx;
+};
+
+static void model_instances_get(uint16_t id, struct shell_model_instance *arr, uint8_t len)
+{
+	const struct bt_mesh_comp *comp = bt_mesh_comp_get();
+	struct bt_mesh_elem *elem;
+	struct bt_mesh_model *mod;
+
+	for (int i = 0; i < len; i++) {
+		elem = bt_mesh_elem_find(comp->elem[i].addr);
+		mod = bt_mesh_model_find(elem, id);
+		if (mod) {
+			arr[i].addr = comp->elem[i].addr;
+			arr[i].elem_idx = mod->elem_idx;
+		}
+	}
+}
+
+static uint8_t str2u8(const char *str)
+{
+	if (isdigit((unsigned char)str[0])) {
+		return strtoul(str, NULL, 0);
+	}
+
+	return (!strcmp(str, "on") || !strcmp(str, "enable"));
+}
+
+bool shell_model_str2bool(const char *str)
+{
+	return str2u8(str);
+}
```

Comment:
These are the duplicates from `subsys/bluetooth/mesh/shell.c`. Could you create a task to make them external (create `subsys/bluetooh/mesh/shell.h`) and use them instead of having duplicates here?